ledger-zcash/src/app.rs

```diff
@@ -478,18 +478,26 @@ where
     pub async fn checkandsign(
         &self,
         data: HsmTxData,
+        tx_version: TxVersion,
     ) -> Result<[u8; 32], LedgerAppError<E::Error>> {
         //this is actually infallible
         let data = data.to_hsm_bytes().unwrap();
-
+        let hex_tx_version = match tx_version {
+            TxVersion::Zip225 => 0x05,
+            TxVersion::Sapling => 0x04,
+            _ => 0x04, //todo: what should the default be?
```

Comment:
Due to it being an enum it represents all the possible values, but the app doesn't support it all, so you should return an error instead of having this default